assets/src/gaussholder.js

```diff
@@ -144,26 +144,22 @@ export default function () {
 	 * Render all placeholders on the page
 	 */
 	return function () {
-		let images = document.getElementsByTagName( 'img' );
-
-		for ( let i = images.length - 1; i >= 0; i-- ) {
-			let img = images[i];
-
-			// Ensure the blank GIF has loaded first
-			if ( img.complete ) {
-				handleElement( img );
-			} else {
-				img.onload = function () {
-					handleElement( this );
-				};
-			}
-		}
+		const images = document.getElementsByTagName( 'img' );
+
+		const options = {
+			rootMargin: '1200px'
+		};
 
-		loadLazily = images;
-		scrollHandler();
+		const imagesObserver = new IntersectionObserver( (entries) => {
+			const visibleImages = entries
+				.filter( ( { isIntersecting } ) => isIntersecting === true );
 
-		if ( loadLazily.length > 0 ) {
-			window.addEventListener( 'scroll', scrollHandler );
-		}
+			visibleImages.forEach( ( { target } ) => loadOriginal( target ) );
+		}, options);
+
+		Array.from( images ).forEach( ( img ) => {
+			imagesObserver.observe( img );
+			handleElement( img );
```

Comment:
I notice that this handle element routine is no longer waiting until the image is loaded. What kinds of cases was the `onload` callback earlier set up to handle? Do we need to account for potentially needing to wait until the blank spacer gif loads?

Reply:
It's funny that you mention this because the plugin wasn't working at all when the bundle was the production one and that's because the development was bringing the source maps and it seems that those ms used to load the sourcemaps are crucial to initialize the images. The production bundle was initialized too early.

I've done some changes for this. The `Gaussholder` function is now called when the DOM is loaded instead and it seems to be working like a charm :)